example/main.go

```diff
@@ -17,23 +17,39 @@
 package main
 
 import (
+	"context"
+
 	"sigs.k8s.io/kube-scheduler-wasm-extension/guest"
-	"sigs.k8s.io/kube-scheduler-wasm-extension/guest/api"
+	"sigs.k8s.io/kube-scheduler-wasm-extension/guest/api/framework"
 )
 
 func main() {
-	guest.Filter = api.FilterFunc(nameEqualsPodSpec)
+	guest.RegisterFilter(&NodeName{})
+}
+
+// NodeName is a plugin that checks if a pod spec node name matches the current node.
+type NodeName struct{}
+
+const (
+	// Name is the name of the plugin used in the plugin registry and configurations.
+	name = "NodeName"
```

Comment:
I think having these as constants only used in one place each adds indirection which reduces readability of the example, which is otherwise simple. I suggest inlining them.